### DISCUSSION TITLE
SSG with some components rendered in SSR

### DISCUSSION THREAD

**Discussion by Arunjenson** [vercel/next.js#61753] [App Router]
consider i have four sections inside a page.jsx where im using SSG but as one of the components is heavy the SSG is taking more time in build so can we opted out of SSG for only that component and opted to SSR for that component and remaining component to be in SSG?

**Comment by hyjoong** [comment]
Are you using the Next.js app router? If so, you can introduce and apply SSR (Server-Side Rendering) and SSG (Static Site Generation) strategies for each server component. When implementing with fetch, if you want to fetch data at build time for a component as in the SSG method, the default value is force-cache, so you don't need to set cache values separately. If you want to fetch data each time a request is made, similar to the SSR method, you can use the no-store attribute.

```javascript
// SSG
fetch('https://~~')

// SSR
fetch('https://...', { cache: 'no-store' })
```

https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating
